OpenRA.Mods.Common/Traits/Render/WithDecoration.cs

```diff
@@ -87,6 +93,16 @@ public void PlaySingleFrame(int frame)
 
 		public IEnumerable<IRenderable> Render(Actor self, WorldRenderer wr)
 		{
+			return !info.SelectionDecoration ? RenderInner(self, wr, self.Bounds) : Enumerable.Empty<IRenderable>();
+		}
+
+		public IEnumerable<IRenderable> RenderAfterWorld(WorldRenderer wr)
+		{
+			return info.SelectionDecoration ? RenderInner(self, wr, self.VisualBounds) : Enumerable.Empty<IRenderable>();
```

Comment:
This makes me thinkg `SelectionDecoration` is really a `should this be rendered at all`, not `only when selected`.